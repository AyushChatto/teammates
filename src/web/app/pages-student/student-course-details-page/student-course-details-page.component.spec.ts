import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { StudentCourseDetailsPageComponent } from './student-course-details-page.component';

describe('StudentCourseDetailsPageComponent', () => {
  let component: StudentCourseDetailsPageComponent;
  let fixture: ComponentFixture<StudentCourseDetailsPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [StudentCourseDetailsPageComponent],
      imports: [
        HttpClientTestingModule,
        RouterTestingModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(StudentCourseDetailsPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with a single course', () => {
    const courseAttributes = {
      createdAt: '1552509888215',
      id: 'CS3281',
      name: 'Thematic Systems',
      timeZone: 'Asia/Singapore',
    };
    component.course = courseAttributes;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
