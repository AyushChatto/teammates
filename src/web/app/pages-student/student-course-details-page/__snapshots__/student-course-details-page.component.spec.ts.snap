// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentCourseDetailsPageComponent should snap with a single course 1`] = `
<tm-student-course-details-page
  Gender={[Function Object]}
  backendUrl={[Function String]}
  course={[Function Object]}
  httpRequestService={[Function HttpRequestService]}
  route={[Function ActivatedRoute]}
  statusMessageService={[Function StatusMessageService]}
  user="undefined"
>
  <h1
    class="page-title"
  >
     Thematic Systems
  </h1><div
    class="section"
  >
    <div
      class="col-sm-12"
    >
      <h4
        class="font-weight-bold"
      >
        Course
      </h4>
    </div>
    <div
      class="card border-primary"
    >
      <div
        class="card-body fill-plain"
      >
        <div
          class="form form-horizontal"
        >
          <div
            class="form-group row"
          >
            <div
              class="col-4 text-right"
            >
              <label>
                Name
              </label>
            </div>
            <div
              class="col-8"
            >
              <span>
                Thematic Systems
              </span>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <div
              class="col-4 text-right"
            >
              <label>
                ID
              </label>
            </div>
            <div
              class="col-8"
            >
              <span>
                CS3281
              </span>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <div
              class="col-4 text-right"
            >
              <label>
                Timezone
              </label>
            </div>
            <div
              class="col-8"
            >
              <span>
                Asia/Singapore
              </span>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <div
              class="col-4 text-right"
            >
              <label>
                Created at
              </label>
            </div>
            <div
              class="col-8"
            >
              <span>
                1552509888215
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</tm-student-course-details-page>
`;

exports[`StudentCourseDetailsPageComponent should snap with default fields 1`] = `
<tm-student-course-details-page
  Gender={[Function Object]}
  backendUrl={[Function String]}
  httpRequestService={[Function HttpRequestService]}
  route={[Function ActivatedRoute]}
  statusMessageService={[Function StatusMessageService]}
  user="undefined"
>
  
</tm-student-course-details-page>
`;
